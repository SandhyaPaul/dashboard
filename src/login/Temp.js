import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';

import { validateToken } from '../redux/app-actions';

class Temp extends React.PureComponent {
	componentWillMount() {
		const token = sessionStorage.getItem('my-access-token');
		const username = sessionStorage.getItem('my-username');
		if (token && username) {
			this.props.validateToken(token, username);
		} else {
			this.props.history.push('/login');
		}
	}

	componentWillReceiveProps(nextProps) {
		if (nextProps.loggedIn === true) {
			this.props.history.push('/app');
		} else {
			this.props.history.push('/login');
		}
	}

	render() {
		return '';
	}
}

export { Temp as TestTemp };

const mapStateToProps = state => {
	return {
		loggedIn: state.app.loggedIn
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators({ validateToken }, dispatch);
};

export { mapStateToProps, mapDispatchToProps };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Temp));
