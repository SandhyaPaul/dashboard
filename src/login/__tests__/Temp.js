import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { TestTemp, mapDispatchToProps, mapStateToProps } from '../Temp';

describe('DOM rendering', () => {
	it('renders LandingPage without breaking', () => {
		shallow(<TestTemp history={[]} />);
	});

	it('DOM snapshot of LandingPage', () => {
		const wrapper = shallow(<TestTemp history={[]} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});

describe('Behavior checks', () => {
	it('redux actions', () => {
		let actions = mapDispatchToProps();
		expect(Object.keys(actions)).toEqual(['validateToken']);
		expect(actions.validateToken).toBeInstanceOf(Function);
	});
	it('gives the right props from state', () => {
		const expectedProps = {
			loggedIn: true
		};
		expect(mapStateToProps({ app: { loggedIn: true } })).toEqual(expectedProps);
	});
	it('goes to login page if nothing found in the storage', () => {
		const history = [];
		const wrapper = shallow(<TestTemp history={history} />);
		expect(history).toEqual(['/login']);
	});
	it('goes to login page if validateToken returned isLoggedIn as false', () => {
		const history = [];
		const validateFn = jest.fn();
		sessionStorage.setItem('my-access-token', 'weird token');
		sessionStorage.setItem('my-username', 'darthvader');
		const wrapper = shallow(<TestTemp history={history} validateToken={validateFn} />);
		expect(validateFn.mock.calls.length).toEqual(1);
		expect(validateFn.mock.calls[0]).toEqual(['weird token', 'darthvader']);
		wrapper.setProps({ loggedIn: false });
		expect(history).toEqual(['/login']);
	});
	it('goes to login page if validateToken returned isLoggedIn as true', () => {
		const history = [];
		const validateFn = jest.fn();
		sessionStorage.setItem('my-access-token', 'weird token');
		sessionStorage.setItem('my-username', 'darthvader');
		const wrapper = shallow(<TestTemp history={history} validateToken={validateFn} />);
		expect(validateFn.mock.calls.length).toEqual(1);
		expect(validateFn.mock.calls[0]).toEqual(['weird token', 'darthvader']);
		wrapper.setProps({ loggedIn: true });
		expect(history).toEqual(['/app']);
	});
});
