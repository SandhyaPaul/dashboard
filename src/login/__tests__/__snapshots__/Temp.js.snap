// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM rendering DOM snapshot of LandingPage 1`] = `""`;
